src/java/boa/compiler/SymbolTable.java

```diff
@@ -435,15 +436,28 @@ else if (aggregators.containsKey(name))
 	}
 
 	public List<Class<?>> getAggregators(final String name, final BoaType type) {
-		final List<Class<?>> aggregators = new ArrayList<Class<?>>();
-
-		if (type instanceof BoaTuple)
-			for (final BoaType subType : ((BoaTuple) type).getTypes())
-				aggregators.add(this.getAggregator(name, subType));
-		else
-			aggregators.add(this.getAggregator(name, type));
+		final List<Class<?>> searchResult = new ArrayList<Class<?>>();
+		if (possibleInBuiltAgg(name)) {
+			if (type instanceof BoaTuple)
+				searchResult.add(this.getAggregator(name, type));
+			else if (type instanceof BoaArray)
+				searchResult.add(this.getAggregator(name, ((BoaArray)type).getType()));
+			else
+				searchResult.add(this.getAggregator(name, type));
+		} else if (this.functions.hasFunction(name)) {
+			searchResult.add(UserDefinedAggregator.class);
+		}
+		return searchResult;
+	}
 
-		return aggregators;
+	private boolean possibleInBuiltAgg(String name) {
+		Set<String> names = aggregators.keySet();
+		for (final String entry: names) {
+			if(entry.contains(name)) {
```

Comment:
should this be contains()? so if I have a built in named 'top' and I try to use an aggregator 'op' it matches? I think you just want a simple `aggregators.containsKey(name)` for this function.